e2e/framework/machineset.go

```diff
@@ -166,12 +168,24 @@ func WaitForMachineSet(cl client.Client, name string) {
 	}, WaitOverLong, RetryMedium).Should(Succeed())
 }
 
-// GetMachineSet gets a machineset by its name from the default machine API namespace.
-func GetMachineSet(cl client.Client, name string) (*clusterv1.MachineSet, error) {
-	machineSet := &clusterv1.MachineSet{}
-	key := client.ObjectKey{Namespace: CAPINamespace, Name: name}
+// GetMachineSet gets a machineset by its name.
+func GetMachineSet(cl client.Client, name string, namespace string) (*clusterv1.MachineSet, error) {
+	if name == "" {
+		return nil, fmt.Errorf("MachineSet name cannot be empty")
+	}
+
+	machineSet := &clusterv1.MachineSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
 
-	Expect(cl.Get(ctx, key, machineSet)).To(Succeed())
+	Eventually(komega.Get(machineSet), time.Minute, RetryShort).Should(SatisfyAny(
+		Succeed(),
+		WithTransform(apierrors.IsNotFound, BeTrue()),
```

Comment:
Again, if we are fetching an object, why would we want to allow it to not be created?

Reply:
For AfterAll clean resource, before I first GetMachineSet then delete, if machineset already was deleted in testing, will return 404, so add this.  I will remove this and update cleanup step, we can DeleteMachineSets directly.